cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1445,10 +1496,10 @@ class ObjectOutputStream final : public io::OutputStream {
   // in the completion handler.
   struct UploadState {
     std::mutex mutex;
-    std::condition_variable cv;
     Aws::Vector<S3Model::CompletedPart> completed_parts;
     int64_t parts_in_progress = 0;
     Status status;
+    Future<> completed = Future<>::MakeFinished(Status::OK());
```

Comment:
Nit: `completed` is a little bit misleading here.  A user could write, flush, and continue writing.  I'm not sure what a better name would be but maybe something like `writes_in_progress`?

Reply:
I renamed it `pending_parts_completed` if that makes more sense.